src/System.Drawing.Common/tests/mono/System.Drawing/ColorTests.cs

```diff
@@ -0,0 +1,183 @@
+            Assert.Equal("Color [ActiveBorder]", Color.FromKnownColor(KnownColor.ActiveBorder).ToString());
+            Assert.Equal("Color [A=1, R=2, G=3, B=4]", Color.FromArgb(1, 2, 3, 4).ToString());
+            Assert.Equal("Color [Empty]", Color.Empty.ToString());
+        }
+
+        [Fact]
+        public void FromName_Invalid()
+        {
+            Color c = Color.FromName("OingoBoingo");
+            Assert.True(c.IsNamedColor);
+            Assert.Equal(c.ToArgb(), 0);
+            Assert.Equal(c.Name, "OingoBoingo");
+        }
+
+        private void CheckRed(string message, Color color)
+        {
+            Assert.Equal(255, color.A);
+            Assert.Equal(255, color.R);
+            Assert.Equal(0, color.G);
+            Assert.Equal(0, color.B);
+            Assert.Equal("Red", color.Name);
+            Assert.True(!color.IsEmpty);
+            Assert.True(color.IsKnownColor);
+            Assert.True(color.IsNamedColor);
+            Assert.True(!color.IsSystemColor);
+        }
+
+        [Fact]
+        public void SerializationRoundtrip()
+        {
+            Color c = Color.Red;
+            CheckRed("original", c);
+
+            BinaryFormatter bf = new BinaryFormatter();
+            MemoryStream ms = new MemoryStream();
+            bf.Serialize(ms, c);
+
+            ms.Position = 0;
+            Color color = (Color)bf.Deserialize(ms);
+            CheckRed("deserialized", color);
+        }
+
+        // serialized under MS 2.0, can be deserialized on both MS 1.1 SP1 and 2.0
+        private static readonly byte[] ColorRed =
+            {
+                0x00, 0x01, 0x00, 0x00, 0x00, 0xFF, 0xFF, 0xFF, 0xFF, 0x01, 0x00, 0x00, 0x00, 0x00,
+                0x00, 0x00, 0x00, 0x0C, 0x02, 0x00, 0x00, 0x00, 0x51, 0x53, 0x79, 0x73, 0x74, 0x65, 0x6D, 0x2E, 0x44,
+                0x72, 0x61, 0x77, 0x69, 0x6E, 0x67, 0x2C, 0x20, 0x56, 0x65, 0x72, 0x73, 0x69, 0x6F, 0x6E, 0x3D, 0x32,
+                0x2E, 0x30, 0x2E, 0x30, 0x2E, 0x30, 0x2C, 0x20, 0x43, 0x75, 0x6C, 0x74, 0x75, 0x72, 0x65, 0x3D, 0x6E,
+                0x65, 0x75, 0x74, 0x72, 0x61, 0x6C, 0x2C, 0x20, 0x50, 0x75, 0x62, 0x6C, 0x69, 0x63, 0x4B, 0x65, 0x79,
+                0x54, 0x6F, 0x6B, 0x65, 0x6E, 0x3D, 0x62, 0x30, 0x33, 0x66, 0x35, 0x66, 0x37, 0x66, 0x31, 0x31, 0x64,
+                0x35, 0x30, 0x61, 0x33, 0x61, 0x05, 0x01, 0x00, 0x00, 0x00, 0x14, 0x53, 0x79, 0x73, 0x74, 0x65, 0x6D,
+                0x2E, 0x44, 0x72, 0x61, 0x77, 0x69, 0x6E, 0x67, 0x2E, 0x43, 0x6F, 0x6C, 0x6F, 0x72, 0x04, 0x00, 0x00,
+                0x00, 0x04, 0x6E, 0x61, 0x6D, 0x65, 0x05, 0x76, 0x61, 0x6C, 0x75, 0x65, 0x0A, 0x6B, 0x6E, 0x6F, 0x77,
+                0x6E, 0x43, 0x6F, 0x6C, 0x6F, 0x72, 0x05, 0x73, 0x74, 0x61, 0x74, 0x65, 0x01, 0x00, 0x00, 0x00, 0x09,
+                0x07, 0x07, 0x02, 0x00, 0x00, 0x00, 0x0A, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x8D, 0x00,
+                0x01, 0x00, 0x0B
+            };
```

Comment:
We have a whole suite of serialization-related tests, that validates roundtripping between core and desktop.  Should this be moved there instead?
cc: @ViktorHofer

Reply:
sorry I can't find the code you pointed to. Question still applies?

Reply:
Nope.  It was addressed with the latest commit. Thanks.